planning/behavior_path_planner/include/behavior_path_planner/utils/path_safety_checker/safety_check.hpp

```diff
@@ -103,7 +103,7 @@ boost::optional<PoseWithVelocityAndPolygonStamped> getInterpolatedPoseWithVeloci
  * @param debug The debug information for collision checking.
  * @return true if distance is safe.
  */
-bool checkCollision(
+std::optional<std::vector<Polygon2d>> checkCollision(
```

Comment:
It seems that it is enough to show that the planned path is safe if the output is empty (not using `std::optional`).

Reply:
@rej55 
The reason `std::optional` is used is because I want to minimize the code changes (on avoidance side and start/goal planner). With std::optional, I only need to remove `!` on both avoidance and start planner.

Reply:
@rej55 
Fixed in d5b99a3ff76c50b788e8fe3c07bc7b119ce6f316